### PR TITLE
API Migrate SilverStripeNagivator classes

### PR DESCRIPTION
**Pull request by GuySartorelli**
These classes are useful with `silverstripe/admin` and `silverstripe/versioned` without needing `silverstripe/cms`.

See also https://github.com/silverstripe/silverstripe-versioned-admin/pull/253 and https://github.com/silverstripe/silverstripe-admin/pull/1340

## Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1339

### CODE REVIEW

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php [hunk 1]
```diff
@@ -16,13 +17,13 @@ public function getHTML()
...
             $this->record->CMSEditLink(),
-            _t('SilverStripe\\CMS\\Controllers\\ContentController.CMS', 'CMS')
+            _t(ContentController::class . '.CMS', 'CMS')
         );
     }
 
     public function getTitle()
     {
-        return _t('SilverStripe\\CMS\\Controllers\\ContentController.CMS', 'CMS', 'Used in navigation. Should be a short label');
+        return _t(ContentController::class . 'CMS', 'CMS', 'Used in navigation. Should be a short label');
```
Comment (dhensby) [inline code/Controllers/SilverStripeNavigatorItem_CMSLink.php]:
```suggestion
        return _t(ContentController::class . '.CMS', 'CMS', 'Used in navigation. Should be a short label');
```

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php [hunk 2]
```diff
@@ -16,13 +17,13 @@ public function getHTML()
         return sprintf(
             '<a href="%s">%s</a>',
             $this->record->CMSEditLink(),
-            _t('SilverStripe\\CMS\\Controllers\\ContentController.CMS', 'CMS')
+            _t(ContentController::class . '.CMS', 'CMS')
```
Comment (dhensby) [inline code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: Should this not just use `$this->getTitle()` - if not, it should use the same fallback text as `getTitle()` (third param: `Used in navigation. Should be a short label`)

**Review by dhensby — DISMISSED** [review]

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php (hunk 2 above)
Comment (ChloeSartorelli) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: I'm not refactoring the class, I'm just migrating some of them and in this case changing a hardcoded string to `::class` syntax.

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php (hunk 1 above)
Comment (ChloeSartorelli) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: Good catch, I'll fix that tomorrow

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php (hunk 2 above)
Comment (dhensby) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: I think if you're going to change LoC, then they should be changed so they are fully correct, not partly changed.

By slipping these unrelated changes into a migration commit/PR you're risking regressions for no benefit. If we at least take the chance to improve the implementation when slipping these unrelated changes in, then that somewhat justifies the risks of regressions slip through the net.

Comment (ChloeSartorelli) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: I'll revert this change tomorrow then.

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php (hunk 1 above)
Comment (ChloeSartorelli) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: Reverted instead per https://github.com/silverstripe/silverstripe-cms/pull/2761#discussion_r970544389

File: code/Controllers/SilverStripeNavigatorItem_CMSLink.php (hunk 2 above)
Comment (ChloeSartorelli) [reply code/Controllers/SilverStripeNavigatorItem_CMSLink.php]: Reverted

**Comment by GuySartorelli** [comment]
Needs https://github.com/silverstripe/silverstripe-admin/pull/1340 to be merged before it will go green.

**Review by sabina-talipova — APPROVED** [review]
Preview works perfect with or without installed silverstripe-cms module.

**Pull request merged**